src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -394,16 +394,25 @@ ModuleCapability GetModuleDataCapabilities(
   return module_data_capabilities;
 }
 
-bool isModuleTypeAndDataMatch(const std::string& module_type,
-                              const smart_objects::SmartObject& module_data) {
+/**
+ * @brief Clears unrelated module data parameters
+ * @param module type in request
+ * @param smart object of module_data
+ * @return true if the correct module parameter is present, false otherwise
+ */
+bool ClearUnrelatedModuleData(const std::string& module_type,
+                              smart_objects::SmartObject& module_data) {
   LOG4CXX_AUTO_TRACE(logger_);
   const auto& all_module_types = RCHelpers::GetModulesList();
```

Comment:
```cpp
auto all_module_types = RCHelpers::GetModulesList();
all_module_types.erase(std::remove(all_module_types.begin(),
                          all_module_types.end(), 
                          module_type), all_module_types.end());
for (const auto& type : all_module_types) {
  const std::string module_key = data_mapping(type);
  if (module_data.keyExists(module_key)) {
    module_data.erase(module_key);
  }
}
return module_data.keyExists(module_key);
```

Reply:
I don't personally see this as being more readable, nor does it seem more efficient than the current version, so I don't see a good reason to make this change.